src/Sarif.Viewer.VisualStudio.Shell.Core/StatusBarService.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Sarif.Viewer.Shell
+{
+    /// <inheritdoc cref="IStatusBarService"/>
+    public class StatusBarService : IStatusBarService
+    {
+        private readonly IServiceProvider serviceProvider;
+        private IVsStatusbar statusBar;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StatusBarService"/> class.
+        /// </summary>
+        /// <param name="serviceProvider">The service provider.</param>
+        public StatusBarService(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+        }
+
+        /// <summary>
+        /// Gets the status bar.
+        /// </summary>
+        /// <value>The status bar.</value>
+        protected IVsStatusbar StatusBar
+        {
+            get
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+
+                if (statusBar == null)
+                {
+                    statusBar = serviceProvider.GetService(typeof(SVsStatusbar)) as IVsStatusbar;
+                    Assumes.Present(statusBar);
+                }
+
+                return statusBar;
+            }
+        }
+
+        /// <inheritdoc cref="IStatusBarService.SetStatusTextAsync(string)"/>
+        public async Task SetStatusTextAsync(string text)
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            this.StatusBar.IsFrozen(out int frozen);
+
+            if (frozen != 0)
+            {
+                this.StatusBar.FreezeOutput(0);
+            }
+
+            this.StatusBar.SetText(text);
```

Comment:
>; [](http://example.com/codeflow?start=40&length=1)

after setting text, should we freeze status bar`this.StatusBar.FreezeOutput(1);`?

I saw the doc below unset/set the freeze
[https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/extensibility/extending-the-status-bar.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/main/docs/extensibility/extending-the-status-bar.md)